app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -138,6 +174,7 @@ def create_vm(args = {})
       if (image_id = args[:image_id])
         args.merge!({:template => image_id})
       end
+      args[:os] = { :type => determine_os_type(args[:foreman_host]) } if supports_operating_systems?
```

Comment:
I think you can pass the host OS and arch to avoid passing a god object to `determine_os_type`.